File: YoutubeClone/DTO/ChannelSummary.cs
namespace YoutubeClone.DTO
{
    public class ChannelSummary
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime DateCreated { get; set; }
    }
}

File: YoutubeClone/ClientApp/src/api/models/channel-summary.ts
export interface ChannelSummary {
    id: string;
    userId: string;
    userName: string;
    userProfilePictureUrl: string;
    name: string;
    description: string;
    thumbnailUrl: string;
    dateCreated: Date;
}

export function mapJsonToChannelSummary(json: any): ChannelSummary {
    return {
        id: json.id,
        userId: json.userId,
        userName: json.userName,
        userProfilePictureUrl: json.userProfilePictureUrl,
        name: json.name,
        description: json.description,
        thumbnailUrl: json.thumbnailUrl,
        dateCreated: new Date(json.dateCreated)
    }
}